---
title: "chipseq"
output:
  html_document:
    toc: yes
---

```{r echo=F,result='hide',error=F,warning=F,message=F}
source('ChIP.shinyrep.helpers.R')
loadGlobalVars()
targets <<- ChIPhelper.init("readTargets")
peaks   <<- ChIPhelper.init("readPeaks")
``` 

## Quality Control ChIPseq ##

### Sequencing quality ###
The sequencing quality of the run was good, and the read distribution over the libraries was good. See tables the SAV quality tables:

```{r echo=F,results='asis',error=F,warning=F,message=F}
# I used to include a already rendered MD file with the bustard statistics, but
# now we can generate it by calling the perl-xslt parser directly from the helpers
#```{r autodoc, child='bustard.md', eval=TRUE}
cat(ChIPhelper.Bustard(),sep="\n")
```

### Mapping
Mapping was done using Bowtie. The mapping statistics show the total number of reads delivered to the aligner, the number of mapped reads, the reads that failed to align, and the number of reads discarded because of mapping to multiple positions:

```{r echo=F,results='asis',error=F,warning=F,message=F}
cat(ChIPhelper.Bowtie(),sep="\n")
``` 

### PCR bottleneck coefficient
The PBC (PCR bottleneck coefficient) is an approximate measure of library complexity. Provisionally, 0-0.5 is severe bottlenecking, 0.5-0.8 is moderate bottlenecking, 0.8-0.9 is mild bottlenecking, while 0.9-1.0 is no bottlenecking.

Very low values can indicate a technical problem, such as PCR bias, or a biological finding, such as a very rare genomic feature. Nuclease-based assays (DNase, MNase) detecting features with base-pair resolution (transcription factor footprints, positioned nucleosomes) are expected to recover the same read multiple times, resulting in a lower PBC score for these assays. Note that the most complex library, random DNA, would approach 1.0, thus the very highest values can indicate technical problems with libraries. Some common numbers from ENCODE datasets are: 82% for TF ChIP, 89% for His ChIP, 77% for DNase, 98% for FAIRE, and 97% for control ENCODE datasets have no or mild bottlenecking.

```{r echo=F,results='asis',error=F,warning=F,message=F}
cat(ChIPhelper.PBC(),sep="\n")
```

### Raw reads qualities, sequence bias and duplication
```{r echo=F,results='asis',error=F,warning=F,message=F}
cat(ChIPhelper.Fastqc(web=F),sep="\n")
```

### IPstrength
To estimate the IP strength, we attempt to decompose the population of IP reads into two distinct components: those pulled down by the antibody, and background.

```{r echo=F,results='asis',error=F,warning=F,message=F}
cat(ChIPhelper.IPstrength(web=F),sep="\n")
```

The plot is useful to detect several forms of biases:
* The distance between the two curves represents the enrichment of the IP versus the input. Close curves means weak enrichment.
* Undersequencing is shown by the curves close to the x-axis for a large percentage of bins, representing no reads were aligned on this bins. 
* Sequencing bias towards a small set of specific regions is shown by the curves close to the y-axis for a large percentage of tags.

### Cross-correlation analysis

**Cross-correlation**
A measure of enrichment derived without dependence on prior determination of enriched regions. Forward and reverse strand read coverage signal tracks are computed (number of unique mapping read starts at each base in the genome on the + and - strand counted separately). The forward and reverse tracks are shifted towards and away from each other by incremental distances and for each shift, the Pearson correlation coefficient is computed. In this way, a cross-correlation profile is computed, representing the correlation between forward and reverse strand coverage at different shifts. The highest cross-correlation value is obtained at a strand shift equal to the predominant fragment length in the dataset as a result of clustering/enrichment of relative fixed-size fragments around the binding sites of the target factor or feature.

**Normalized Strand Cross-correlation coefficient (NSC):**
The NSC is the ratio of the maximal cross-correlation value (which occurs at strand shift equal to fragment length) divided by the background cross-correlation (minimum cross-correlation value over all possible strand shifts). Higher values indicate more enrichment, values less than 1.1 are relatively low NSC scores, and the minimum possible value is 1 (no enrichment). This score is sensitive to technical effects; for example, high-quality antibodies such as H3K4me3 and CTCF score well for all cell types and ENCODE production groups, and variation in enrichment in particular IPs is detected as stochastic variation. This score is also sensitive to biological effects; narrow marks score higher than broad marks (H3K4me3 vs H3K36me3, H3K27me3) for all cell types and ENCODE production groups, and features present in some individual cells, but not others, in a population are expected to have lower scores.

**Relative Strand Cross-correlation coefficient (RSC):**
The RSC is the ratio of the fragment-length cross-correlation value minus the background cross-correlation value, divided by the phantom-peak cross-correlation value minus the background cross-correlation value. The minimum possible value is 0 (no signal), highly enriched experiments have values greater than 1, and values much less than 1 may indicate low quality.

```{r echo=F,results='asis',error=F,warning=F,message=F}
cat(ChIPhelper.PhantomPeak(web=F),sep="\n")
```

## Peaks called

A summary of the peaks called in the different comparisons is shown here:

```{r echo=F,results='asis',error=F,warning=F,message=F}
peaks <- ChIPhelper.init("readPeaks")

l <- lapply(peaks,function(x) {
	x <- table(x$chr)
	xx <- as.numeric(x)
	names(xx) <- names(x)
	xx
})
p <- Reduce(function(x,y) {
	z <- merge(x,y,all=T,by=0)
	rownames(z) <- z[,1]
	z[,-1]
},l)
#colnames(p) <- 1:ncol(p)
colnames(p) <- names(l)

# print the legend
#cat("",fill=T)
#cat(kable(data.frame(column=1:length(l),sample=names(l))),sep="\n")
#cat("",fill=T)

# print the peaks table
cat("",fill=T)
cat(kable(p,row.names=T),sep="\n")
cat("",fill=T)
```
