---
title: "SHINYREPS_PROJECT"
output:
  html_document:
    toc: yes
    css: styles.css
---

<div class="contentbox">

```{r Initialise, echo=F,result='hide',error=F,warning=F,message=F}
source('DE.shinyrep.helpers.R')
loadGlobalVars()
load(SHINYREPS_DE_EDGER,envir=.GlobalEnv)		# the outcome from the DE analysis
DEhelper.init("prepareDEdataTable")
DEhelper.init("renderUcscGeneLinks")
DEhelper.init("prepareDistanceMatrix")
``` 

## Quality Control RNAseq ##

### Sequencing quality ###
The sequencing quality of the run was good, and the read distribution over the libraries was good. See tables the SAV quality tables:

```{r Bustard_paragraph, echo=F,results='asis',error=F,warning=F,message=F}
# I used to include a already rendered MD file with the bustard statistics, but
# now we can generate it by calling the perl-xslt parser directly from the helpers
#```{r autodoc, child='bustard.md', eval=TRUE}
cat(DEhelper.Bustard(),sep="\n")
```

### Raw reads qualities, sequence bias and duplication
```{r FastQC_paragraph, echo=F,results='asis',error=F,warning=F,message=F}
cat(DEhelper.Fastqc(web=F),sep="\n")
```

### Mapping
Mapping was done using STAR. The program version and genome assembly are described in the following table. The following parameters were used to get only uniquely mapping reads:

```{r STAR_parameters_paragraph, echo=F,results='asis',error=F,warning=F,message=F}
cat(DEhelper.STARparms(),sep="\n")
``` 

The mapping statistics show the total number of reads delivered to the aligner, the number of uniquely mapped reads, and the number of reads discarded because of mapping to multiple positions or not mapping at all to the reference genome.

```{r STAR_paragraph, echo=F,results='asis',error=F,warning=F,message=F}
cat(DEhelper.STAR(),sep="\n")
``` 

### PCR duplication assessment
Measuring the fraction of duplicated reads is a common quality control step for NGS data sets, which can hint towards too low library complexity due to too many PCR cycles during library preparation or other problems in the sequencing workflow. However in RNA-Seq duplicate reads arise naturally due to highly expressed genes, which renders the overall read duplication rate useless.

```{r Dupradar_paragraph, echo=F,results='asis',error=F,warning=F,message=F}
cat(DEhelper.dupRadar(web=F),sep="\n")
```

### Subread
Mapping statistics of reads assigned to known features in the genome and evaluated in further steps. In addition, the count and rate of reads is shown that could not be assigned to the investigated genomic features due to various reasons detailed in the column headers.

```{r Subread_paragraph, echo=F,results='asis',error=F,warning=F,message=F}
cat(DEhelper.Subread(),sep="\n")
```

### RNA types
Check for an enrichment of specific RNA types. This plots helps in detecting if the extraction protocol worked well, and the expected RNA types were sequenced.

```{r RNAtypes_paragraph, echo=F,results='asis',error=F,warning=F,message=F}
cat(DEhelper.RNAtypes(web=F),sep="\n")
```

### Gene body coverage
Check if reads coverage is uniform and if there is any 5' or 3' bias:

```{r GeneBodyCoverage_paragraph, echo=F,results='asis',error=F,warning=F,message=F}
cat(DEhelper.geneBodyCov(web=F),sep="\n")
```

### Batch effect
The MDS plot shows the sample relations based on multidimensional scaling (MDS). Distances in the plot can be interpreted in terms of leading log2 fold change. Overall, sample replicates should show little distances between them, while groups should show higher distances between them. If the ratio between/within groups distances is high, there is no batch effect (or technical variation) if the experiment was well randomized:

```{r MDSplot_paragraph, echo=F,error=F,warning=F,message=F,fig.align="center"}
DEhelper.MDS()		# sample MDS 
```

Clustering the samples based on the distance calculated on the expression levels of the 50 most variable genes, one can have an idea which samples have similar gene expression profiles. This, together with the MDS plot, enables us to detect if the replicates behave similar in terms of expression and spot batches of samples related by other characteristics rather than condition: <br>

```{r Heatmap1_paragraph, echo=F,error=F,warning=F,message=F,fig.align="center"}
DEhelper.cluster()	# heatmap of top variant 'n' genes of the counts-per-milion table
```

Finally, sample to sample correlation based on the distance calculated on the expression profile gives us a measure of sample similarity: <br>

```{r Heatmap2_paragraph, echo=F,error=F,warning=F,message=F,fig.align="center"}
DEhelper.corr()		# heatmap of sample to sample distances
```

## Differential Expression Analysis

```{r DifferentialExpression_paragraph, echo=F,results='asis',error=F,warning=F,message=F,fig.align="center"}
for(i in 1:length(colnames(conts))) {

	# subsection
	cat("### ",colnames(conts)[i],fill=T)	
	cat("",fill=T)
	
	# MA plot
	fdr <- if(exists("input")) { fdr <- if(is.numeric(input$slider_pval)) input$slider_pval else .01 } else .01
	genes <- DEhelper.DEgenes(i)
	cat("There are",sum(genes$FDR < fdr),
		"genes highlighted in the MA plot with corrected pvalue (Benjamini and Hochberg, FDR) <",fdr,fill=T)
	DEhelper.MAplot(i,fdr)	# should be fine at taking the value from the UI slider
	cat("\n",fill=T)

	# and DE table
	cat("The table contains the top 25 differentially expressed genes in terms of fold change and FDR:",fill=T)
	cat("",fill=T)
	genes$PValue <- format(genes$PValue,scientific=TRUE)
	genes$FDR <- format(genes$FDR,scientific=TRUE)
#	cat(kable(genes[1:25,-1],row.names=T,align="r"),sep="\n")	# show only the top 25 genes
	cat(kable(genes[1:25,-4],row.names=T,align="r"),sep="\n")	# show only the top 25 genes
	cat("",fill=T)
}
```

## Used tools and versions for this analysis

```{r ToolVersions_paragraph, echo=F,results='asis',error=F,warning=F,message=F}

# this section requires an init() that loads the location to the *_LOG variables 

table.content <- data.frame(  tool=c(
									  "FastQC",
									  "STAR",
									  "Samtools",
									  "Subread",
									  "Picard",
									  "GeneBodyCoverage",
									  "R/EdgeR",
									  "R/DEseq2"
									  ),
							  version=c(
									  Toolhelper.VersionReporter("FastQC",   SHINYREPS_FASTQC_LOG ),
									  Toolhelper.VersionReporter("STAR",     SHINYREPS_STAR_LOG ),
									  Toolhelper.VersionReporter("Samtools", SHINYREPS_BAMINDEX_LOG),
									  Toolhelper.VersionReporter("Subread",  SHINYREPS_SUBREAD_LOG),
									  Toolhelper.VersionReporter("Picard",   SHINYREPS_BAM2BW_LOG),
									  Toolhelper.VersionReporter("GeneBodyCoverage", SHINYREPS_GENEBODYCOVERAGE_LOGS),
									  Toolhelper.VersionReporter("R/EdgeR",  SHINYREPS_EDGER_LOGS ),
									  Toolhelper.VersionReporter("R/DEseq2", SHINYREPS_DESEQ_LOGS)
									  )
						   )
kable(table.content)

```

</div>

