---
title: "SHINYREPS_PROJECT"
output:
  html_document:
    toc: yes
---

```{r echo=F,result='hide',error=F,warning=F,message=F}
source('variant.shinyrep.helpers.R')
loadGlobalVars()
# load(SHINYREPS_DE_EDGER,envir=.GlobalEnv)		# the outcome from the DE analysis
# VARhelper.init("prepareDEdataTable")
# VARhelper.init("renderUcscGeneLinks")
# VARhelper.init("prepareDistanceMatrix")
``` 

## Quality Control RNAseq ##

### Sequencing quality ###
The sequencing quality of the run was good, and the read distribution over the libraries was good.

### Raw reads qualities, sequence bias and duplication
```{r echo=F,results='asis',error=F,warning=F,message=F}
cat(VARhelper.Fastqc(web=F),sep="\n")
```

### Mapping
The mapping statistics show the total number of reads delivered to the aligner, the number of mapped reads and reads not mapping at all to the ref aligning to the wrong reference genome.

```{r echo=F,results='asis',error=F,warning=F,message=F}
cat(VARhelper.BWA(),sep="\n")
```

### Variant calling
Variant calling can be performed using GATK UnifiedGenotyper and GATK HaplotypeCaller.
Both tools include default read filters that exclude reads from variant calling.

#### GATK UnifiedGenotyper filtered read statistics
```{r echo=F,results='asis',error=F,warning=F,message=F}
cat(VARhelper.GATKug(),sep="\n")
```


## Used tools and versions for this analysis

```{r echo=F,results='asis',error=F,warning=F,message=F}
cat(VARhelper.ToolVersions(),sep="\n")
``` 
